### ISSUE TITLE
:presence considers a key not present if value is the empty set or nil

### ISSUE THREAD

**Issue by pjlegato** [mylesmegyesi/metis#5]
A `:presence` validator designed to test for the presence of a key will fail the map if the value of that key is `nil` or `#{}` (but `false` works). 

Since the key is present, it should pass validation, even if its value is `#{}` or `nil` -- `(keys {:foo nil})` and `(keys {:foo #{}})` both return `(:foo)`.

For example:

```
(defvalidator :has-key-foo
  [:foo :presence])
user> (has-key-foo {:foo 123})
{}
nil
user> (has-key-foo {:foo #{}})
{:foo ["must be present"]}
nil
user> (has-key-foo {:foo false})
{}
nil
user> (has-key-foo {:foo nil})
{:foo ["must be present"]}
nil
```

**Comment by mylesmegyesi** [comment]
The intention of this validator was to test not only that the key is present in the map, but also that it is _logically_ present.

Since metis is inspired by ActiveModel, this validator behaves similarly to the ActiveModel presence validator, which simply calls the [`blank?`](http://api.rubyonrails.org/classes/Object.html#method-i-blank-3F) method under the hood.

Sorry for any confusion.

**Comment by pjlegato** [comment]
The problem is that nil or an empty set might well be a semantically valid value in certain cases. Could we at least note this point in the README, so other people won't be confused?

**Comment by mylesmegyesi** [comment]
@pjlegato I've added a new validator to help address your use case. I believe the new [`:contains`](https://github.com/mylesmegyesi/metis/wiki/contains) validator should suit your needs. It has been released in version 0.3.3.

**Comment by pjlegato** [comment]
Great, thanks!
